components/sidebar/sidebar_category/sidebar_category_menu/index.ts

```diff
@@ -2,42 +2,40 @@
 // See LICENSE.txt for license information.
 
 import {connect} from 'react-redux';
-import {Dispatch, bindActionCreators} from 'redux';
+import {Dispatch, bindActionCreators, ActionCreatorsMapObject} from 'redux';
 
-import Permissions from 'mattermost-redux/constants/permissions';
+import {renameCategory, deleteCategory} from 'mattermost-redux/actions/channel_categories';
 import {getCurrentTeam} from 'mattermost-redux/selectors/entities/teams';
-import {haveITeamPermission} from 'mattermost-redux/selectors/entities/roles';
+import {ActionFunc} from 'mattermost-redux/types/actions';
 import {ChannelCategory} from 'mattermost-redux/types/channel_categories';
 import {GlobalState} from 'mattermost-redux/types/store';
 
-import SidebarCategoryMenu from './sidebar_category_menu';
+import {createCategory} from 'actions/views/channel_sidebar';
 
-type OwnProps = {
-    category: ChannelCategory;
-}
+import SidebarCategoryMenu from './sidebar_category_menu';
 
 function makeMapStateToProps() {
-    return (state: GlobalState, ownProps: OwnProps) => {
+    return (state: GlobalState) => {
         const currentTeam = getCurrentTeam(state);
 
-        let canCreatePublicChannel = false;
-        let canCreatePrivateChannel = false;
-
-        if (currentTeam) {
-            canCreatePublicChannel = haveITeamPermission(state, {team: currentTeam.id, permission: Permissions.CREATE_PUBLIC_CHANNEL});
-            canCreatePrivateChannel = haveITeamPermission(state, {team: currentTeam.id, permission: Permissions.CREATE_PRIVATE_CHANNEL});
-        }
-
         return {
-            canCreatePrivateChannel,
-            canCreatePublicChannel,
+            currentTeamId: currentTeam.id,
         };
     };
 }
 
+type Actions = {
+    createCategory: (teamId: string, displayName: string, channelIds?: string[] | undefined) => {data: ChannelCategory};
+    deleteCategory: (categoryId: string) => void;
+    renameCategory: (categoryId: string, displayName: string) => void;
+}
+
 function mapDispatchToProps(dispatch: Dispatch) {
     return {
-        actions: bindActionCreators({
+        actions: bindActionCreators<ActionCreatorsMapObject<ActionFunc>, Actions>({
```

Comment:
This is done because i'm using `void` as the return type instead of `ActionFunc` for a few functions, which I don't think the un-connected component needs to care about, since the actions are 'fire-and-forget'.

There also seems to be an issue with how it interprets `createCategory`, this is the error I get:
```
Type '{ createCategory: (teamId: string, displayName: string, channelIds?: string[] | undefined) => { data: ChannelCategory; }; deleteCategory: (categoryId: string) => void; renameCategory: (categoryId: string, newName: string) => void; }' is not assignable to type '{ createCategory: (teamId: string, displayName: string, channelIds?: string[] | undefined) => (dispatch: DispatchFunc) => Promise<ActionResult>; deleteCategory: (categoryId: string) => ActionFunc; renameCategory: (categoryId: string, displayName: string) => ActionFunc; }'.ts(2345)
```

For some reason it's including the `dispatch` function wrapper when coming out of the binding. I think the `ActionCreatorsMapObject` is actually the correct way we're supposed to be doing this anyways.

Reply:
It just seems ridiculous that we'd need to specify all the types here. That might just come down to our type definitions in mattermost-redux being broken though. Something to investigate later, I guess.